File: 705. design-hashset/main.cpp
class MyHashSet {
public:
    /** Initialize your data structure here. */
    MyHashSet() {

    }
    
    void add(int key) {
        _hashmapValues.insert(pair<int, char>(key, ' '));
    }
    
    void remove(int key) {
        _hashmapValues.erase(key);
    }
    
    /** Returns true if this set contains the specified element */
    bool contains(int key) {
        return _hashmapValues.find(key) != _hashmapValues.end();
    }
private:
    unordered_map<int, char> _hashmapValues;

};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * MyHashSet* obj = new MyHashSet();
 * obj->add(key);
 * obj->remove(key);
 * bool param_3 = obj->contains(key);
 */

File: 705. design-hashset/main.js
/**
 * Initialize your data structure here.
 */
 var MyHashSet = function() {
    this.hash_table = {};
};

/** 
 * @param {number} key
 * @return {void}
 */
MyHashSet.prototype.add = function(key) {
    this.hash_table[key] = true;
};

/** 
 * @param {number} key
 * @return {void}
 */
MyHashSet.prototype.remove = function(key) {
    delete this.hash_table[key];
    // 或者：this.hash_table[key] = false;
};

/**
 * Returns true if this set contains the specified element 
 * @param {number} key
 * @return {boolean}
 */
MyHashSet.prototype.contains = function(key) {
    return Boolean(this.hash_table[key]);
};

/**
 * Your MyHashSet object will be instantiated and called as such:
 * var obj = new MyHashSet()
 * obj.add(key)
 * obj.remove(key)
 * var param_3 = obj.contains(key)
 */